Unit test TestStatefulSetLifecycle is very flaky

**Describe the bug**
<!--
A clear and concise description of what the bug is.
If you believe this bug is a security issue, please don't use this template and follow our [security guidelines](/SECURITY.md)
-->

TestStatefulSetLifecycle failed very frequently after merging https://github.com/antrea-io/antrea/pull/3072
```
--- FAIL: TestStatefulSetLifecycle (1.11s)
    antrea_ipam_controller_test.go:110: 
        	Error Trace:	antrea_ipam_controller_test.go:110
        	            				antrea_ipam_controller_test.go:189
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestStatefulSetLifecycle
```

https://github.com/tnqn/antrea/runs/7684550323?check_suite_focus=true
https://github.com/antrea-io/antrea/runs/7689295086?check_suite_focus=true
https://github.com/antrea-io/antrea/runs/7687228906?check_suite_focus=true

--- a/build/charts/antrea/templates/controller/clusterrole.yaml
+++ b/build/charts/antrea/templates/controller/clusterrole.yaml
@@ -270,6 +270,7 @@ rules:
       - ippools/status
     verbs:
       - update
+      - patch
   - apiGroups:
       - apps
     resources:

--- a/build/yamls/antrea-aks.yml
+++ b/build/yamls/antrea-aks.yml
@@ -3514,6 +3514,7 @@ rules:
       - ippools/status
     verbs:
       - update
+      - patch
   - apiGroups:
       - apps
     resources:

--- a/build/yamls/antrea-eks.yml
+++ b/build/yamls/antrea-eks.yml
@@ -3514,6 +3514,7 @@ rules:
       - ippools/status
     verbs:
       - update
+      - patch
   - apiGroups:
       - apps
     resources:

--- a/build/yamls/antrea-gke.yml
+++ b/build/yamls/antrea-gke.yml
@@ -3514,6 +3514,7 @@ rules:
       - ippools/status
     verbs:
       - update
+      - patch
   - apiGroups:
       - apps
     resources:

--- a/build/yamls/antrea-ipsec.yml
+++ b/build/yamls/antrea-ipsec.yml
@@ -3527,6 +3527,7 @@ rules:
       - ippools/status
     verbs:
       - update
+      - patch
   - apiGroups:
       - apps
     resources:

--- a/build/yamls/antrea.yml
+++ b/build/yamls/antrea.yml
@@ -3514,6 +3514,7 @@ rules:
       - ippools/status
     verbs:
       - update
+      - patch
   - apiGroups:
       - apps
     resources:

--- a/pkg/controller/ipam/antrea_ipam_controller.go
+++ b/pkg/controller/ipam/antrea_ipam_controller.go
@@ -19,6 +19,7 @@ package ipam
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"strings"
 	"time"
@@ -27,6 +28,7 @@ import (
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	apitypes "k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/informers"
@@ -395,11 +397,16 @@ func (c *AntreaIPAMController) updateIPPoolCounters(poolName string) error {
 		return nil
 	}
 
-	ipPoolToUpdate := ipPool.DeepCopy()
-	ipPoolToUpdate.Status.Usage.Used = used
-	ipPoolToUpdate.Status.Usage.Total = total
+	patch, _ := json.Marshal(map[string]interface{}{
+		"status": map[string]interface{}{
+			"usage": map[string]interface{}{
+				"used":  used,
+				"total": total,
+			},
+		},
+	})
 
-	_, err = c.crdClient.CrdV1alpha2().IPPools().UpdateStatus(context.TODO(), ipPoolToUpdate, metav1.UpdateOptions{})
+	_, err = c.crdClient.CrdV1alpha2().IPPools().Patch(context.TODO(), ipPool.Name, apitypes.MergePatchType, patch, metav1.PatchOptions{}, "status")
 	if err != nil {
 		return fmt.Errorf("failed to update IPPool %s counters, error: %v", poolName, err)
 	}

--- a/pkg/controller/ipam/antrea_ipam_controller_test.go
+++ b/pkg/controller/ipam/antrea_ipam_controller_test.go
@@ -31,7 +31,6 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes/fake"
-	"k8s.io/klog/v2"
 
 	crdv1a2 "antrea.io/antrea/pkg/apis/crd/v1alpha2"
 	fakecrd "antrea.io/antrea/pkg/client/clientset/versioned/fake"
@@ -145,48 +144,49 @@ func TestStatefulSetLifecycle(t *testing.T) {
 	}
 
 	for _, tt := range tests {
-		stopCh := make(chan struct{})
-		defer close(stopCh)
+		t.Run(tt.name, func(t *testing.T) {
+			stopCh := make(chan struct{})
+			defer close(stopCh)
 
-		klog.InfoS("Running", "test", tt.name)
-		namespace, pool, statefulSet := initTestObjects(!tt.dedicatedPool, tt.dedicatedPool, tt.replicas)
-		crdClient := fakecrd.NewSimpleClientset(pool)
-		k8sClient := fake.NewSimpleClientset(namespace, statefulSet)
+			namespace, pool, statefulSet := initTestObjects(!tt.dedicatedPool, tt.dedicatedPool, tt.replicas)
+			crdClient := fakecrd.NewSimpleClientset(pool)
+			k8sClient := fake.NewSimpleClientset(namespace, statefulSet)
 
-		informerFactory := informers.NewSharedInformerFactory(k8sClient, 0)
+			informerFactory := informers.NewSharedInformerFactory(k8sClient, 0)
 
-		crdInformerFactory := crdinformers.NewSharedInformerFactory(crdClient, 0)
-		poolInformer := crdInformerFactory.Crd().V1alpha2().IPPools()
-		poolLister := poolInformer.Lister()
+			crdInformerFactory := crdinformers.NewSharedInformerFactory(crdClient, 0)
+			poolInformer := crdInformerFactory.Crd().V1alpha2().IPPools()
+			poolLister := poolInformer.Lister()
 
-		controller := NewAntreaIPAMController(crdClient, informerFactory, crdInformerFactory)
-		require.NotNil(t, controller)
-		informerFactory.Start(stopCh)
-		crdInformerFactory.Start(stopCh)
+			controller := NewAntreaIPAMController(crdClient, informerFactory, crdInformerFactory)
+			require.NotNil(t, controller)
+			informerFactory.Start(stopCh)
+			crdInformerFactory.Start(stopCh)
 
-		go controller.Run(stopCh)
+			go controller.Run(stopCh)
 
-		var allocator *poolallocator.IPPoolAllocator
-		var err error
-		// Wait until pool propagates to the informer
-		pollErr := wait.PollImmediate(100*time.Millisecond, 1*time.Second, func() (bool, error) {
-			allocator, err = poolallocator.NewIPPoolAllocator(pool.Name, crdClient, poolLister)
-			if err != nil {
-				return false, nil
-			}
-			return true, nil
+			var allocator *poolallocator.IPPoolAllocator
+			var err error
+			// Wait until pool propagates to the informer
+			pollErr := wait.PollImmediate(100*time.Millisecond, 3*time.Second, func() (bool, error) {
+				allocator, err = poolallocator.NewIPPoolAllocator(pool.Name, crdClient, poolLister)
+				if err != nil {
+					return false, nil
+				}
+				return true, nil
+			})
+			require.NoError(t, pollErr)
+			defer allocator.ReleaseStatefulSet(statefulSet.Namespace, statefulSet.Name)
+
+			// Verify create event was handled by the controller
+			verifyPoolAllocatedSize(t, pool.Name, poolLister, tt.expectAllocatedSize)
+
+			// Delete StatefulSet
+			k8sClient.AppsV1().StatefulSets(namespace.Name).Delete(context.TODO(), statefulSet.Name, metav1.DeleteOptions{})
+
+			// Verify Delete event was processed
+			verifyPoolAllocatedSize(t, pool.Name, poolLister, 0)
 		})
-		require.NoError(t, pollErr)
-		defer allocator.ReleaseStatefulSet(statefulSet.Namespace, statefulSet.Name)
-
-		// Verify create event was handled by the controller
-		verifyPoolAllocatedSize(t, pool.Name, poolLister, tt.expectAllocatedSize)
-
-		// Delete StatefulSet
-		k8sClient.AppsV1().StatefulSets(namespace.Name).Delete(context.TODO(), statefulSet.Name, metav1.DeleteOptions{})
-
-		// Verify Delete event was processed
-		verifyPoolAllocatedSize(t, pool.Name, poolLister, 0)
 	}
 }
 

